### PR TITLE
[Plugin] Creating smarter system for queuing up tasks performed in training vs inference

### PR DESCRIPTION
**Pull request by zmhassan**
Depends-On: https://github.com/AICoE/log-anomaly-detector/pull/180

## Description

When building your own model or performing inference there is often custom steps involved. Some steps may be skipped depending on the scenario. It would be good to be able to queue up tasks and perform them.

### CODE REVIEW

File: anomaly_detector/jobs/tasks.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        try:
+            self.w2v_model.load(self.storage_adapter.W2V_MODEL_PATH)
+        except ModelLoadException as ex:
+            logging.error("Failed to load W2V model: %s" % ex)
+            self.update_w2v_model = False
+            self.recreate_models = True
+
+    def execute(self):
+        """Train models for anomaly detection."""
+        data, _ = self.storage_adapter.load_data(config_type="train")
```
[the remark this answers is not in the document]
Comment (zmhassan) [reply anomaly_detector/jobs/tasks.py]: This should be setup in the constructer of the Facade. If we are training we should detect if this is ES storage or local file. Then intialize the storage attribute.

File: anomaly_detector/jobs/tasks.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+class AbstractCommand(metaclass=ABCMeta):
+    """Generic command that gets executed when added to the TaskManager steps."""
+
+    @abstractmethod
+    def execute(self):
+        """Generic method that gets executed by the task manager."""
+        raise NotImplementedError("Please implement the <execute method>")
+
+
+class SomTrainCommand(AbstractCommand):
```
[the remark this answers is not in the document]
Comment (zmhassan) [reply anomaly_detector/jobs/tasks.py]: Yes. I can add that in.

[the remark this answers is not in the document]
Comment (zmhassan) [reply anomaly_detector/jobs/tasks.py]: @durandom Now tests are using it. for Training. I'll add in the inference and double step train-infer loop as separate commands in separate PR's.

[the remark this answers is not in the document]
Comment (zmhassan) [reply anomaly_detector/jobs/tasks.py]: This PR is now integrated with tests and is ready for review.

**Comment by zmhassan** [comment]
@durandom This PR is ready for review. I've addressed all issues.

**Comment by goern** [comment]
recheck

**Comment by goern** [comment]
@zmhassan what is the 'pr-in-review' label used for?

**Comment by zmhassan** [comment]
> @zmhassan what is the 'pr-in-review' label used for?

Oh thats just a label I put when I'm finished the work and waiting

**Pull request merged**